drivers/chibios/serial_usart_duplex.c

```diff
@@ -0,0 +1,306 @@
+#include <stdatomic.h>
+#include "quantum.h"
+#include "serial.h"
+#include "printf.h"
+
+#include "ch.h"
+#include "hal.h"
+
+#if !defined(USART_CR1_M0)
+#    define USART_CR1_M0 USART_CR1_M  // some platforms (f1xx) dont have this so
+#endif
+
+#if !defined(USE_GPIOV1)
+// The default PAL alternate modes are used to signal that the pins are used for USART
+#    if !defined(SERIAL_USART_TX_PAL_MODE)
+#        define SERIAL_USART_TX_PAL_MODE 7
+#    endif
+#    if !defined(SERIAL_USART_RX_PAL_MODE)
+#        define SERIAL_USART_RX_PAL_MODE 7
+#    endif
+#endif
+
+#if !defined(SERIAL_USART_DRIVER)
+#    define SERIAL_USART_DRIVER UARTD1
+#endif
+
+#if !defined(SERIAL_USART_CR1)
+#    define SERIAL_USART_CR1 (USART_CR1_PCE | USART_CR1_PS | USART_CR1_M0)  // parity enable, odd parity, 9 bit length
+#endif
+
+#if !defined(SERIAL_USART_CR2)
+#    define SERIAL_USART_CR2 (USART_CR2_STOP_1)  // 2 stop bits
+#endif
+
+#if !defined(SERIAL_USART_CR3)
+#    define SERIAL_USART_CR3 0x0
+#endif
+
+#if !defined(SERIAL_USART_TX_PIN)
+#    define SERIAL_USART_TX_PIN A9
+#endif
+
+#if !defined(SERIAL_USART_RX_PIN)
+#    define SERIAL_USART_RX_PIN A10
+#endif
+
+#if defined(USART1_REMAP)
+#    define USART_REMAP (AFIO->MAPR |= AFIO_MAPR_USART1_REMAP);
+#elif defined(USART2_REMAP)
+#    define USART_REMAP (AFIO->MAPR |= AFIO_MAPR_USART2_REMAP);
+#elif defined(USART3_PARTIALREMAP)
+#    define USART_REMAP (AFIO->MAPR |= AFIO_MAPR_USART3_REMAP_PARTIALREMAP);
+#elif defined(USART3_FULLREMAP)
+#    define USART_REMAP (AFIO->MAPR |= AFIO_MAPR_USART3_REMAP_FULLREMAP);
+#endif
+
+#if !defined(SELECT_SERIAL_SPEED)
+#    define SELECT_SERIAL_SPEED 1
+#endif
+
+#if defined(SERIAL_USART_SPEED)
```

Comment:
A lot of this looks to be identical, or very close to the half duplex driver.  It may be worth pulling out the shared config and putting them in an h file to be shared

Reply:
I have moved all shared configurations into serial_usart.h and added QMK copyright headers.